llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -295,15 +296,95 @@ saveResultModules(std::vector<std::unique_ptr<Module>> &ResModules) {
   return Res;
 }
 
+// Each fallback device library corresponds to one bit in "require mask" which
+// is an unsigned int32. getDeviceLibBit checks which fallback device library
+// is required for FuncName and returns the corresponding bit. The corresponding
+// mask for each fallback device library is:
+// fallback-cassert:      0x1
+// fallback-cmath:        0x2
+// fallback-cmath-fp64:   0x4
+// fallback-complex:      0x8
+// fallback-complex-fp64: 0x10
+static uint32_t getDeviceLibBits(const std::string &FuncName) {
+
+  static constexpr uint32_t DeviceLibAssert = 0x1;
+  static constexpr uint32_t DeviceLibCmath = 0x2;
+  static constexpr uint32_t DeviceLibCmath64 = 0x4;
+  static constexpr uint32_t DeviceLibComplex = 0x8;
+  static constexpr uint32_t DeviceLibComplex64 = 0x10;
+  if (FuncName == "__devicelib_assert_fail") {
+    return DeviceLibAssert;
+  }
+  size_t Len =
```

Comment:
suggest to factor out the repeated pattern into a function

Reply:
Hi, @kbobrovs 
The repeated patter has been removed since we don't use sorted array + binary_search any more.